nifi-nar-bundles/nifi-evtx-bundle/nifi-evtx-processors/src/test/java/org/apache/nifi/processors/evtx/ParseEvtxTest.java

```diff
@@ -0,0 +1,481 @@
+        Record record3 = mock(Record.class);
+        RootNode rootNode1 = mock(RootNode.class);
+        RootNode rootNode2 = mock(RootNode.class);
+        RootNode rootNode3 = mock(RootNode.class);
+        ProcessSession session = mock(ProcessSession.class);
+        FlowFile flowFile = mock(FlowFile.class);
+        FlowFile created1 = mock(FlowFile.class);
+        FlowFile updated1 = mock(FlowFile.class);
+        FlowFile created2 = mock(FlowFile.class);
+        FlowFile updated2 = mock(FlowFile.class);
+        FlowFile created3 = mock(FlowFile.class);
+        FlowFile updated3 = mock(FlowFile.class);
+        MalformedChunkException malformedChunkException = new MalformedChunkException("Test", null, offset, chunkNum, badChunk);
+
+        when(session.create(flowFile)).thenReturn(created1).thenReturn(created2).thenReturn(created3).thenReturn(null);
+
+        when(session.write(eq(created1), any(OutputStreamCallback.class))).thenAnswer(invocation -> {
+            ((OutputStreamCallback) invocation.getArguments()[1]).process(out);
+            return updated1;
+        });
+
+        when(session.write(eq(created2), any(OutputStreamCallback.class))).thenAnswer(invocation -> {
+            ((OutputStreamCallback) invocation.getArguments()[1]).process(out2);
+            return updated2;
+        });
+
+        when(session.write(eq(created3), any(OutputStreamCallback.class))).thenAnswer(invocation -> {
+            ((OutputStreamCallback) invocation.getArguments()[1]).process(out3);
+            return updated3;
+        });
+
+        when(record1.getRootNode()).thenReturn(rootNode1);
+        when(record2.getRootNode()).thenReturn(rootNode2);
+        when(record3.getRootNode()).thenReturn(rootNode3);
+
+        when(fileHeader.hasNext()).thenReturn(true).thenReturn(true).thenReturn(true).thenReturn(false);
+        when(fileHeader.next()).thenThrow(malformedChunkException).thenReturn(chunkHeader1).thenReturn(chunkHeader2).thenReturn(null);
+
+        when(chunkHeader1.hasNext()).thenReturn(true).thenReturn(false);
+        when(chunkHeader1.next()).thenReturn(record1).thenReturn(null);
+
+        when(chunkHeader2.hasNext()).thenReturn(true).thenReturn(true).thenReturn(false);
+        when(chunkHeader2.next()).thenReturn(record2).thenReturn(record3).thenReturn(null);
+
+        parseEvtx.processRecordGranularity(session, componentLog, flowFile, basename, in);
+
+        verify(malformedChunkHandler).handle(flowFile, session, parseEvtx.getName(basename, chunkNum, null, ParseEvtx.EVTX_EXTENSION), badChunk);
+        verify(rootNodeHandler1).handle(rootNode1);
+        verify(rootNodeHandler1).close();
+        verify(rootNodeHandler2).handle(rootNode2);
+        verify(rootNodeHandler2).close();
+        verify(rootNodeHandler3).handle(rootNode3);
+        verify(rootNodeHandler3).close();
+    }
+
+    @Test
+    public void fileGranularityLifecycleTest() throws IOException, ParserConfigurationException, SAXException {
```

Comment:
@mattyb149 lifecycle tests of the 3 different granularities using a doctored evtx file that has good chunks, bad chunk headers, and a malformed record